documentation/docs/98-configuration.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Configuration
+---
+
+Your project's configuration lives in a `svelte.config.cjs` file. All values are optional. The complete list of options, with defaults, is shown here:
+
+```js
+/** @type {import('@sveltejs/kit').Config} */
+module.exports = {
+	// options passed to svelte.compile (https://svelte.dev/docs#svelte_compile)
+	compilerOptions: null,
+
+	// an array of file extensions that should be treated as Svelte components
+	extensions: ['.svelte'],
+
+	kit: {
+		adapter: null,
+		amp: false,
+		appDir: '_app',
+		files: {
+			assets: 'static',
+			lib: 'src/lib',
+			routes: 'src/routes',
+			serviceWorker: 'src/service-worker',
+			setup: 'src/setup',
+			template: 'src/app.html'
+		},
+		host: null,
+		hostHeader: null,
+		paths: {
+			assets: '',
+			base: ''
+		},
+		prerender: {
+			crawl: true,
+			enabled: true,
+			force: false,
+			pages: ['*']
+		},
+		startGlobal: null,
+		target: null
+	},
+
+	// options passed to svelte.preprocess (https://svelte.dev/docs#svelte_preprocess)
+	preprocess: null
+};
+```
+
+#### adapter
+
+Determines how the output of `svelte-kit build` is converted for different platforms. Can be specified as a `string` or a `[string, object]` tuple if you need to pass options.
```

Comment:
Using arrays as tiles tuples always seems ugly to me. I don't know what I'd suggest instead. Maybe if you pass an object instead of a string, there's one key that's used as that adapter name, and then the whole object is the options?

Reply:
Agree. I always hate arrays used in this manner as well

Reply:
I don't love it but it seems preferable to the alternatives, a single objects would be confusing. There is also some precedent for this approach.

Reply:
See https://github.com/sveltejs/kit/issues/439. I think the tuple thing is gross, and we should do it the way we do Rollup plugins, but was hesitant because I didn't want to make all the adapters CJS (which we'd need to do until we can use ESM config files) so I kicked the can down the road

Reply:
:+1: If this is one of the APIs we're cool with messing with before 1.0, this sounds fine for me for now as a description of how things currently work.